src/torrent.cpp

```diff
@@ -9246,9 +9246,9 @@ namespace libtorrent
 			return;
 
 		// now, pick one of the rarest pieces to download
-		int const pick = random(std::uint32_t(rarest_pieces.size() - 1));
+		int const pick = int(random(aux::numeric_cast<std::uint32_t>(rarest_pieces.end_index() - 1)));
 		bool const was_finished = is_finished();
-		m_picker->set_piece_priority(rarest_pieces[pick], 1);
+		m_picker->set_piece_priority(rarest_pieces[pick], default_piece_priority);
```

Comment:
good catch. there's another call on line 9218 that probably should use ``default_piece_priority`` instead of 1.

I will back-port these to 1.1